browser/src/map/Clipboard.js

```diff
@@ -876,55 +879,60 @@ L.Clipboard = L.Class.extend({
 		// that command so we are sure the clipboard is set before
 		// fetching it.
 
-		const that = this;
-
-		if (that._commandCompletion.length > 0)
-			window.app.console.error('Already have ' + that._commandCompletion.length +
+		if (this._commandCompletion.length > 0)
+			window.app.console.error('Already have ' + this._commandCompletion.length +
 						 ' pending clipboard command(s)');
 
-		const url = that.getMetaURL() + '&MimeType=text/html,text/plain;charset=utf-8';
-
-		// Share a single fetch
-		var fetchPromise = function() {
-			return new Promise((resolve, reject) => {
-				try {
-					var result = fetch(url).then(response => response.text());
-					resolve(result);
-				} catch (err) {
-					reject(err);
-				}
+		const url = this.getMetaURL() + '&MimeType=text/html,text/plain;charset=utf-8';
+		const commandCompletePromise = new Promise((resolve, reject) => {
+			window.app.console.log('New ' + command + ' promise');
+			// FIXME: add a timeout cleanup too ...
+			this._commandCompletion.push({
+				resolve: resolve,
+				reject: reject,
 			});
-		};
+		});
+
+		await commandCompletePromise;
```

Comment:
why it is called here, and no params?

Reply:
I could, I guess, change the way this worked to look a little more conventional if I made a function that sent the command and returned a promise like this at the same time... maybe that would be easier to grok